src/main/java/com/google/api/codegen/ProtocGeneratorMain.java

```diff
@@ -0,0 +1,125 @@
+
+import com.google.api.codegen.gapic.GapicGeneratorApp;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.tools.ToolOptions;
+import com.google.api.tools.framework.tools.ToolUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import com.google.protobuf.compiler.PluginProtos;
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorRequest;
+import com.google.protobuf.compiler.PluginProtos.CodeGeneratorResponse;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Entrypoint for protoc-plugin invoked generation. Protoc passes input via std.in as a serialized
+ * CodeGeneratorRequest, and expects to read a CodeGeneratorResponse from std.out.
+ */
+public class ProtocGeneratorMain {
+
+  private static final ArtifactType DEFAULT_ARTIFACT_TYPE = GAPIC_CODE;
+
+  public static void main(String[] args) {
+    // TODO(andrealin): Implement. Handle I/O interfacing with protoc.
+  }
+
+  @VisibleForTesting
+  // Parses the InputStream for a CodeGeneratorRequest and returns the generated output in a
+  // CodeGeneratorResponse.
+  public static CodeGeneratorResponse generate(CodeGeneratorRequest request) {
+    try {
+      ToolOptions toolOptions = parseOptions(request);
+
+      GapicGeneratorApp codeGen = new GapicGeneratorApp(toolOptions, DEFAULT_ARTIFACT_TYPE, true);
+
+      codeGen.run();
+      CodeGeneratorResponse response = codeGen.getCodeGeneratorProtoResponse();
+      if (response == null) {
+        throw new RuntimeException(collectDiags(codeGen));
+      }
+      return response;
+    } catch (Exception e) {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      e.printStackTrace(pw);
+      pw.flush();
+      return PluginProtos.CodeGeneratorResponse.newBuilder().setError(sw.toString()).build();
+    }
+  }
+
+  private static ToolOptions parseOptions(CodeGeneratorRequest request) throws Exception {
+    List<FileDescriptorProto> fileDescriptorProtoList = request.getProtoFileList();
+    FileDescriptorSet descriptorSet =
+        FileDescriptorSet.newBuilder().addAllFile(fileDescriptorProtoList).build();
+
+    // Write out DescriptorSet to temp file.
+    File descriptorSetFile;
+
+    descriptorSetFile = File.createTempFile("api", ".desc");
+    FileOutputStream fileoutput = new FileOutputStream(descriptorSetFile);
+    descriptorSet.writeTo(fileoutput);
+    fileoutput.close();
+    descriptorSetFile.deleteOnExit();
+
+    List<String> parsedArgs = new LinkedList<>();
+    parsedArgs.add("--descriptor_set");
+    parsedArgs.add(descriptorSetFile.getAbsolutePath());
+
+    // For now, assume there will only be one proto package to be generated.
```

Comment:
If we are making this assumption for now, we should error out in the case that there are additional files.